server/memphis_handler_schemas.go

```diff
@@ -221,104 +169,46 @@ func (sh SchemasHandler) CreateNewSchema(c *gin.Context) {
 
 	if err != nil {
 		serv.Warnf(err.Error())
-		c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
+		c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": err.Error()})
 		return
 	}
-	schemaContent := strings.ToLower(body.SchemaContent)
-	err = validateSchemaContent(schemaContent)
+	schemaContent := body.SchemaContent
+	err = validateSchemaContent(schemaContent, schemaType)
 	if err != nil {
 		serv.Warnf(err.Error())
-		c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
+		c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": err.Error()})
 		return
 	}
-	filter := bson.M{"name": schemaName}
-	var schema models.Schema
-	err = schemasCollection.FindOne(context.TODO(), filter).Decode(&schema)
-	if err != nil && !strings.Contains(err.Error(), "no documents") {
-		serv.Errorf("CreateSchema error: " + err.Error())
-		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
-		return
-	}
-	versionNumber := len(schema.Versions) + 1
-	var active bool
-	if versionNumber == 1 {
-		active = true
-	} else {
-		active = false
-	}
 	newSchemaVersion := models.SchemaVersion{
 		ID:            primitive.NewObjectID(),
-		VersionNumber: versionNumber,
-		Active:        active,
+		VersionNumber: 1,
+		Active:        true,
 		CreatedByUser: user.Username,
 		CreationDate:  time.Now(),
 		SchemaContent: schemaContent,
 	}
 	newSchema := models.Schema{
-		ID:        primitive.NewObjectID(),
-		Name:      schemaName,
-		Type:      schemaType,
-		Versions:  schema.Versions,
-		IsDeleted: false,
+		ID:       primitive.NewObjectID(),
+		Name:     schemaName,
+		Type:     schemaType,
+		Versions: []primitive.ObjectID{newSchemaVersion.ID},
 	}
 
-	filter = bson.M{"name": newSchema.Name, "is_deleted": false}
-	update := bson.M{
-		"$setOnInsert": bson.M{
-			"_id":  newSchema.ID,
-			"name": newSchema.Name,
-			"type": newSchema.Type,
-		},
-		"$push": bson.M{"versions": newSchemaVersion.ID},
-	}
-	opts := options.Update().SetUpsert(true)
-	updateResults, err := schemasCollection.UpdateOne(context.TODO(), filter, update, opts)
+	_, err = schemasCollection.InsertOne(context.TODO(), newSchema)
```

Comment:
the upsert was the right way to do this one, talk to me in case you don't understand why, so change it back to updateOne with upsert insetad of insertOne